tests/cypress/support/commands/select2.js

```diff
@@ -0,0 +1,73 @@
+/**
+ * ---------------------------------------------------------------------
+ *
+ * GLPI - Gestionnaire Libre de Parc Informatique
+ *
+ * http://glpi-project.org
+ *
+ * @copyright 2015-2024 Teclib' and contributors.
+ * @copyright 2003-2014 by the INDEPNET Development Team.
+ * @licence   https://www.gnu.org/licenses/gpl-3.0.html
+ *
+ * ---------------------------------------------------------------------
+ *
+ * LICENSE
+ *
+ * This file is part of GLPI.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * ---------------------------------------------------------------------
+ */
+
+Cypress.Commands.add('getSelect2DropdownByValue', {prevSubject: 'optional'}, (
+    subject,
+    value
+) => {
+    if (subject) {
+        return cy.wrap(subject).findByRole('combobox', {name: value});
+    } else {
+        return cy.findByRole('combobox', {name: value});
+    }
+});
+
+Cypress.Commands.add('setSelect2Value', {prevSubject: true}, (
+    subject,
+    new_value
+) => {
+    cy.wrap(subject).click();
+    cy.findByRole('option', {name: new_value}).click();
+});
+
+Cypress.Commands.add('select2ValueShouldBeSelected', {prevSubject: 'optional'}, (
+    subject,
+    value
+) => {
+    if (subject) {
+        cy.wrap(subject).getSelect2DropdownByValue(value).should('exist');
+    } else {
+        cy.getSelect2DropdownByValue(value).should('exist');
+    }
+});
+
+Cypress.Commands.add('select2ValueShouldNotBeSelected', {prevSubject: 'optional'}, (
+    subject,
+    value
+) => {
+    if (subject) {
+        cy.wrap(subject).getSelect2DropdownByValue(value).should('not.exist');
+    } else {
+        cy.getSelect2DropdownByValue(value).should('not.exist');
+    }
+});
```

Comment:
IMHO, it would be preferable to have a `getSelect2DropdownByLabel` function and to make `prevSubject` mandatory for `select2ValueShouldBeSelected` and `select2ValueShouldNotBeSelected` functions.

For instance, if you have a form with 2 dropdown, one withe an empty default valeu and the other with a non empty default value, what would this test be able to validate?
```js
cy.get("@form").select2ValueShouldBeSelected("-----");
cy.get("@form").select2ValueShouldNotBeSelected("-----");
```

For this case, the test should look like this:
```js
cy.get("@form").getSelect2DropdownByLabel('Dropdown 1').select2ValueShouldBeSelected("-----");
cy.get("@form").getSelect2DropdownByLabel('Dropdown 2').select2ValueShouldNotBeSelected("-----");
```

Reply:
The main issue is that select2 does not provide the correct accessibility values to it simply (the accessible name of the dropdown is its current value, NOT its label...).

That is why I went with this simpler solution for now, where you select the area of the form that contains your label and then inside this area look directly for the select2 value.

----

But with more code and a bit of hacking we could create a working `getSelect2DropdownByLabel` command.
We can get the actual hidden dropdown using the `findByLabel` method, which probably have some kind of data attribute that can help us look for real select2 item.

In this case I think the method name should not reference select2 at all: maybe just `getDropdownByLabel` ?
The idea would be that if we change library later on we only need to edit this command to not break the tests.